src/llvm_ir_generator.cpp

```diff
@@ -261,11 +267,26 @@ void LLVMIRGenerator::Visit(const RecordVarDeclNode& record_var_decl) {
   for (auto i = std::size_t{0}, e = record_var_decl.inits.size(),
             slot_count = record_type.SlotCount();
        i < slot_count && i < e; ++i) {
+    // TODO: consider the case for union when there are multiple designators. It
+    // seems that both Clang and gcc have initialized the following union with
+    // the value 6.
+    // union shape s = {.circle = 5, .triangle = 6};
     auto& init = record_var_decl.inits.at(i);
+    record_des_vals.clear();
     init->Accept(*this);
     auto init_val = val_recorder.ValOfPrevExpr();
 
-    auto res_addr = builder_.CreateStructGEP(type, base_addr, i);
+    llvm::Value* res_addr = nullptr;
+    if (record_des_vals.empty()) {
+      res_addr = builder_.CreateStructGEP(type, base_addr, i);
+    } else {
+      // TODO: consider array index designators.
+      // NOTE: Only consider identifer designators for now.
+      assert(record_des_vals.size() == 1);
```

Comment:
Is the size of `record_des_vals` ever able to have size greater than 1? Since we always handle initialization of a single member at a time (`auto& init = record_var_decl.inits.at(i);`), I guess the answer is no?

Reply:
> Is the size of `record_des_vals` ever able to have size greater than 1? 

Yes, it is possible for `record_des_vals`'s size to be greater than 1. For example,

```
struct birth {
    int date;
    int month;
    int year;
  };

  struct birth bd1[3] = {
    [2].year = 1995,
    [1].date = 3,
    [0].month = 4,
  };
```

In the above example, `bd1` is an array of `struct birth`. `[2].year` designates the last struct object's year member, which consists of a `ArrDesNode` with index 2 and a `IdDesNode` with a `year` identifier. However, I haven't considered the case with array designator. It gets more complicated with 2D or 3D array, but I don't want to get that far yet. 😕 

> Since we always handle initialization of a single member at a time (auto& init = record_var_decl.inits.at(i);), I guess the answer is no?

I think you mixed up `record_var_decl.inits.at(i)` and `record_des_vals`. In the above example, `auto& init = record_var_decl.inits.at(0);`'s `init` refers this line `[2].year = 1995`, which consists of a vector of `DesNode` and a `InitExprNode`. `record_des_vals` is the vector of `DesNode`, which has a `ArrDesNode` with index 2 and a `IdDesNode` with `year` identifier. I hope this resolves your question. 😄

Reply:
Oops, let me summarize to make sure I've got this right. `record_des_vals` can have a size greater than `1` if the designator is nested. In the example above, `[2].year`, `[2]` would be at index `0`, and `year` would be at index `1`. Is that correct? 😊

Reply:
Yes! That's correct!